test/runtime: remove already covered FQDN tests

tklauser opened the pull request:
Remove three more tests from the runtime FQDN test suite that are either already covered by existing tests or are of limited use.

    test/runtime: remove RuntimeAgentFQDNPolicies default-deny test

    The policy used in the test doesn't have a toPorts.rules.dns rule, so
    the DNS proxy isn't covered at all in the test.
    
    This test is thus testing the default-deny behavior which is already
    covered elsewhere, e.g. various Cilium CLI connectivity tests.
---
    test/runtime: remove RuntimeAgentFQDNPolicies Implement matchPattern * test

    matchPattern * is already covered in existing Cilium CLI connectivity
    tests, though not for subdomains currently.  However, the correct
    application of the provided pattern to FQDNs is already covered in
    various unit tests in pkg/fqdn. Thus the corresponding runtime FQDN test
    case is redundant and can be removed.
---
    test/runtime: remove RuntimeAgentFQDNPolicies DNSSEC test

    The existing FQDN runtime test only checks for the existence of an
    RRSIG record in the DNS response[^1]. The Cilium DNS proxy doesn't touch
    RRSIG records at all. So it looks like the existing test is
    already of limited use and can be removed without replacement.
    
    [^1]: https://github.com/cilium/dnssec-client/blob/master/client.py#L30-L32

For #37838

bimmlerd commented on cilium-cli/connectivity/builder/manifests/client-egress-only-dns-world.yaml:
```diff
@@ -0,0 +1,18 @@
+        dns:
+        - matchPattern: "*.cilium.io"
+    toCIDR:
+      - {{.ExternalDNSServerIP}}/32
```
The `/32` locks us into IPv4, but the option looks like we could also provide IPv6 - Should we parse and then fully qualify the IP in Go instead of doing it via the template?

tklauser replied:
Yeah, good point. The test are currently failing on dual-stack/IPv6-only clusters anyway so I'll need to look into that.

bimmlerd commented on cilium-cli/connectivity/tests/dns.go:
```diff
@@ -0,0 +1,80 @@
+
+			// DNSSEC query (pod to external DNS)
+			t.NewAction(p, fmt.Sprintf("dig-dnssec-%d", i), &client, target, ipFam).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.DigCommand(dnsServer, ipFam, p.targetName, "+dnssec"))
```
I'm not sure I understand - does `dig +dnssec` fail if the `targetName` domain has no DNSSEC enabled? Does `www.cilium.io` have DNSSEC enabled?

tklauser replied:
~It checks for a DNSSEC record for `targetName`, so yes it fails if DNSSEC is not enabled for the given domain. That's what the existing runtime FQDN test currently also does. My interpretation of that test is that it's validating that DNS responses containing DNSSEC records get handled correctly by the DNS proxy.~

Ok, I was wrong about it failing if no DNSSEC record exists (and cilium.io doesn't have an RRSIG record). So the new test is not testing anything wrt. DNSSEC atm 🤦

bimmlerd commented on cilium-cli/connectivity/check/result.go:
```diff
@@ -210,6 +210,13 @@ var (
 		ExitCode: ExitCurlHTTPError,
 	}
+
+	// ResultDNSDrop expects a dropped flow while and a successful command.
```
nit: comment does not parse

bimmlerd reviewed (commented):
I'm not sure I fully get what we're trying to assert re DNSSEC and whether www.cilium.io actually has DNSSEC enabled

tklauser committed:
test/runtime: remove RuntimeAgentFQDNPolicies DNSSEC test

The existing FQDN runtime test only checks for the existence of an
RRSIG record in the DNS response[^1]. The Cilium DNS proxy doesn't touch
RRSIG records at all. So it looks like the existing test is
already of limited use and can be removed without replacement.

[^1]: https://github.com/cilium/dnssec-client/blob/master/client.py#L30-L32

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
test/runtime: remove RuntimeAgentFQDNPolicies Implement matchPattern * test

matchPattern * is already covered in existing Cilium CLI connectivity
tests, though not for subdomains currently.  However, the correct
application of the provided pattern to FQDNs is already covered in
various unit tests in pkg/fqdn. Thus the corresponding runtime FQDN test
case is redundant and can be removed.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

bimmlerd reviewed (approved):
> The Cilium DNS proxy doesn't seem to touch RRSIG records at all. 

Could it be that the intent of the test is verifying that the DNS proxy doesn't drop DNSSEC records? I think the upstream DNS servers of www.cilium.io serve RRSIGs (i.e. root and authoritative NS) so if we drop those that could be a problem. But I think this is easily testable at the unit/component level (and may already be covered).

tklauser committed:
test/runtime: remove RuntimeAgentFQDNPolicies default-deny test

The policy used in the test doesn't have a toPorts.rules.dns rule, so
the DNS proxy isn't covered at all in the test.

This test is thus testing the default-deny behavior which is already
covered elsewhere, e.g. various Cilium CLI connectivity tests.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
